### PR TITLE
adding PL & lite with `torch`

### PR DESCRIPTION
**Pull request by Borda**
## Before submitting

- [x] Was this discussed/approved via a GitHub issue? (no need for typos and docs improvements)
- [x] Did you create/update your **configuration file**?
- [x] Did you **set `runtimes` in config** for GitHub action integration?
- [x] Did you **add your config to CI** in Azure pipeline (only projects with 100+ GitHub stars)?
- [ ] Are all integration **tests passing**?

## What does this PR do? \[optional\]

Adding compatibility check for future PyTorch release

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
pending on https://github.com/Lightning-AI/lightning/pull/15904

**Comment by Borda** [comment]
pending on https://github.com/Lightning-AI/lightning/pull/15904

File: configs/Lightning-AI/pl-lite_torch-develop.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - "--pre"
+
+testing:
+  # additional pytest arguments
+  pytest_args: --strict
+
+runtimes:
+  - {os: "ubuntu-22.04", python: "3.10"}
+  - {os: "macOS-12", python: "3.10"}
+#  - {os: "windows-2022", python: "3.10"}
```
Comment (justusschock) [inline configs/Lightning-AI/pl-lite_torch-develop.yaml:33]: what's the issue here?

File: .azure/ci-testig-parameterized.yml [hunk 2]
```diff
@@ -17,6 +17,8 @@ jobs:
 - template: testing-template.yml
   parameters:
     configs:
+#    - "Lightning-AI/pl_torch-develop.yaml"  # TODO
+#    - "Lightning-AI/pl-lite_torch-develop.yaml"  # TODO
```
Comment (justusschock) [inline .azure/ci-testig-parameterized.yml:21]: will this be a follow-up PR?

File: .github/workflows/ci_testing.yml [hunk 3]
```diff
@@ -48,7 +48,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix: ${{ fromJSON(needs.check-diff.outputs.matrix) }}
-    timeout-minutes: 35
+    timeout-minutes: 90
```
Comment (justusschock) [inline .github/workflows/ci_testing.yml:51]: why did it increase so much? Or is this just some higher value? if so, can we try to find out the actual maximum so we know when we have a regression?

File: configs/Lightning-AI/pl-lite_torch-develop.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  # define installing package extras
+  install_extras: dev
+  # copy some  test from the target repository
+  copy_tests:
+    - tests/tests_lite
+
+contact:
+  slack:
+    # Adrian Wälchli
+    - URC536Z0W
```
Comment (justusschock) [inline configs/Lightning-AI/pl-lite_torch-develop.yaml:14]: could you include me here as well?

File: configs/Lightning-AI/pl_torch-develop.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  # define installing package extras
+  install_extras: dev
+  # copy some  test from the target repository
+  copy_tests:
+    - tests/tests_pytorch
+
+contact:
+  slack:
+    # Carlos Mocholí
+    - U019R2ZS25T
```
Comment (justusschock) [inline configs/Lightning-AI/pl_torch-develop.yaml:14]: could you add me here as well? :)

File: configs/Lightning-AI/pl_torch-develop.yaml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      - "--pre"
+
+testing:
+  # additional pytest arguments
+  pytest_args: --strict
+
+runtimes:
+  - {os: "ubuntu-22.04", python: "3.10"}
+  - {os: "macOS-12", python: "3.10"}
+#  - {os: "windows-2022", python: "3.10"}
```
Comment (justusschock) [inline configs/Lightning-AI/pl_torch-develop.yaml:33]: why no windows?

**Review by justusschock — APPROVED** [review]

File: configs/Lightning-AI/pl-lite_torch-develop.yaml (hunk 1 above)
Comment (Borda) [reply configs/Lightning-AI/pl-lite_torch-develop.yaml:33]: I don't think there was an issue, just copied the existing config and adjusted/pruned :otter:

File: .github/workflows/ci_testing.yml (hunk 3 above)
Comment (Borda) [reply .github/workflows/ci_testing.yml:51]: we never run full TM tests here only integrations to PL (5min top) and from TM we know that the full test suit on Mac takes about 75min :(

File: configs/Lightning-AI/pl_torch-develop.yaml (hunk 6 above)
Comment (Borda) [reply configs/Lightning-AI/pl_torch-develop.yaml:35]: again just copied and let's keep this future integration simple, but if you want I am fine to enable it right now :)

Comment (justusschock) [reply configs/Lightning-AI/pl_torch-develop.yaml:35]: I'd prefer to have a full matrix for everything :) Just so we make sure we don't miss anything while we update it. It shouldn't be too much as we only have one python and one torch version per OS

File: configs/Lightning-AI/pl-lite_torch-develop.yaml (hunk 1 above)
Comment (justusschock) [reply configs/Lightning-AI/pl-lite_torch-develop.yaml:35]: let's enable it  then :)

File: configs/Lightning-AI/pl_torch-develop.yaml (hunk 6 above)
Comment (Borda) [reply configs/Lightning-AI/pl_torch-develop.yaml:35]:
```suggestion
  - {os: "windows-2022", python: "3.10"}
```

File: configs/Lightning-AI/pl-lite_torch-develop.yaml (hunk 1 above)
Comment (Borda) [reply configs/Lightning-AI/pl-lite_torch-develop.yaml:35]:
```suggestion
  - {os: "windows-2022", python: "3.10"}
```

File: configs/Lightning-AI/pl-lite_torch-develop.yaml [hunk 7]
```diff
@@ -0,0 +1,35 @@
+target_repository:
+  HTTPS: https://github.com/Lightning-AI/lightning.git
+  # checkout a particular branch or a tag
+  checkout: master
+  # define installing package extras
+  install_extras: dev
+  # copy some  test from the target repository
+  copy_tests:
+    - tests/tests_lite
```
Comment (carmocca) [inline configs/Lightning-AI/pl-lite_torch-develop.yaml:9]: These have just been renamed to fabric

File: configs/Lightning-AI/pl-lite_torch-develop.yaml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+contact:
+  slack:
+    # Adrian Wälchli
+    - URC536Z0W
+    # Justus Schock
+    - U010XFPFETH
+
+env:
+  # let's limit the installation only to Pytorch-Lightning
+  PACKAGE_NAME: lite
```
Comment (carmocca) [inline configs/Lightning-AI/pl-lite_torch-develop.yaml:20]:
```suggestion
  PACKAGE_NAME: fabric
```

File: configs/Lightning-AI/pl-lite_torch-develop.yaml [hunk 9]
```diff
@@ -0,0 +1,35 @@
+target_repository:
```
Comment (carmocca) [inline configs/Lightning-AI/pl-lite_torch-develop.yaml:1]: Update the filename too

**Review by carmocca — APPROVED** [review]

File: configs/Lightning-AI/pl-lite_torch-develop.yaml (hunk 7 above)
Comment (Borda) [reply configs/Lightning-AI/pl-lite_torch-develop.yaml:9]: let's rename it back when it lands on master or we can enable this testing from custom branch :)

Comment (carmocca) [reply configs/Lightning-AI/pl-lite_torch-develop.yaml:9]: It has landed already

File: configs/Lightning-AI/fabric_torch-develop.yaml [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+contact:
+  slack:
+    # Adrian Wälchli
+    - URC536Z0W
+    # Justus Schock
+    - U010XFPFETH
+
+env:
+  # let's limit the installation only to Pytorch-Lightning
```
Comment (akihironitta) [inline configs/Lightning-AI/fabric_torch-develop.yaml:19]:
```suggestion
  # let's limit the installation only to Fabric
```

**Review by akihironitta — APPROVED** [review]

**Pull request closed without merging**